internal/cmd/kafka/command_cluster_update.go

```diff
@@ -66,31 +74,97 @@ func (c *clusterCommand) update(cmd *cobra.Command, args []string, prompt form.P
+				return err
+			}
+			update.Spec.DisplayName = nil
+		}
 	}
 
-	if cmd.Flags().Changed("cku") {
-		cku, err := cmd.Flags().GetUint32("cku")
+	if cmd.Flags().Changed("availability") {
+		availability, err := cmd.Flags().GetString("availability")
 		if err != nil {
 			return err
 		}
-		updatedCku, err := c.validateResize(int32(cku), &currentCluster, prompt)
-		if err != nil {
-			return err
+		if availability == "" {
+			return errors.New("`--availability` must not be empty")
 		}
-		update.Spec.Config = &cmkv2.CmkV2ClusterSpecUpdateConfigOneOf{CmkV2Dedicated: &cmkv2.CmkV2Dedicated{Kind: "Dedicated", Cku: updatedCku}}
+		update.Spec.SetAvailability(strings.ToUpper(strings.ReplaceAll(availability, "-", "_")))
 	}
 
-	updatedCluster, err := c.V2Client.UpdateKafkaCluster(clusterID, update)
+	// Both `--type` and `--cku` require changing the config
+	if cmd.Flags().Changed("type") || cmd.Flags().Changed("cku") {
+		// Set base config before setting type or CKU
+		updatedClusterType := getClusterType(currentCluster.Spec.GetConfig())
+		if cmd.Flags().Changed("type") {
+			clusterType, err := cmd.Flags().GetString("type")
+			if err != nil {
+				return err
+			}
+			updatedClusterType = clusterType
+		}
+		switch updatedClusterType {
+		case skuBasic:
+			update.Spec.SetConfig(cmkv2.CmkV2ClusterSpecUpdateConfigOneOf{CmkV2Basic: new(cmkv2.CmkV2Basic)})
+		case skuStandard:
+			update.Spec.SetConfig(cmkv2.CmkV2ClusterSpecUpdateConfigOneOf{CmkV2Standard: new(cmkv2.CmkV2Standard)})
+		case skuDedicated:
+			update.Spec.SetConfig(cmkv2.CmkV2ClusterSpecUpdateConfigOneOf{CmkV2Dedicated: new(cmkv2.CmkV2Dedicated)})
+		default:
+			return fmt.Errorf(`unsupported cluster type "%s"`, updatedClusterType)
+		}
+
+		if cmd.Flags().Changed("type") {
+			clusterType, err := cmd.Flags().GetString("type")
+			if err != nil {
+				return err
+			}
+			if clusterType == "" {
+				return errors.New("`--type` must not be empty")
+			}
+			switch clusterType {
+			case skuBasic:
+				update.Spec.Config.CmkV2Basic.SetKind("Basic")
+			case skuStandard:
+				update.Spec.Config.CmkV2Standard.SetKind("Standard")
+			case skuDedicated:
+				update.Spec.Config.CmkV2Dedicated.SetKind("Dedicated")
+			}
+		}
+
+		if cmd.Flags().Changed("cku") {
+			cku, err := cmd.Flags().GetUint32("cku")
+			if err != nil {
+				return err
+			}
+			switch updatedClusterType {
+			case skuBasic, skuStandard:
+				return fmt.Errorf(errors.ClusterResizeNotSupportedErrorMsg)
+			case skuDedicated:
+				updatedCku, err := c.validateResize(int32(cku), &currentCluster, prompt)
+				if err != nil {
+					return err
+				}
+				update.Spec.Config.CmkV2Dedicated.SetCku(updatedCku)
+			}
+		}
+	}
+
+	updatedCluster, err := c.V2Client.UpdateKafkaCluster(clusterId, update)
 	if err != nil {
-		return errors.NewWrapErrorWithSuggestions(err, "failed to update Kafka cluster", errors.KafkaClusterUpdateFailedSuggestions)
+		return err
```

Comment:
Is there a reason why we are no longer returning a suggestion?

Reply:
The suggestion is "A cluster can't be updated while still provisioning." but we return an error earlier if that's the case, so this would just lead users down the wrong path.

Reply:
I see, is there any other errors that would cause this to suggest looking into? If not, no worries, but this may help oncalls while debugging